src/pamiq_core/model/interface.py

```diff
@@ -86,3 +89,31 @@ def forward(self, *args, **kwds) -> Any:
     def __call__(self, *args, **kwds) -> Any:
         """Calls `forward` method."""
         return self.forward(*args, **kwds)
+
+    def sync(self) -> None:
+        """Copies params of training model to self._inference_model if
+        needed."""
+        if self._need_sync:
+            self.sync_model(self.inference_model)
+
+    @property
+    def _need_sync(self) -> bool:
+        """Return whether It is necessary to synchronize training model and
+        inference model."""
+        return self.has_inference_model and (not self.inference_only)
+
+    def sync_model(self, inference_model: T) -> None:
+        """Copies params of training model to self._inference_model if needed.
+
+        Example:
+            class TrainingTorchModel(TrainingModel[InferenceTorchModel]):
+                @override
+                def sync_model(self, inference_model: InferenceTorchModel) -> None:
+                    with torch.no_grad():
+                        for inference_param, training_param in zip(
+                            inference_model.model.parameters(),
+                            self.model.parameters(),
+                        ):
+                            inference_param.copy_(training_param)
+        """
+        raise NotImplementedError
```

Comment:
Discordで話した通り。このレベルのシンプルさが基底クラスでは求められる。
https://discord.com/channels/982974075410460722/1319517296846176327/1345568058269237288
https://discord.com/channels/982974075410460722/1319517296846176327/1345569031062556712
```suggestion
    @abstractmethod
    def sync(self) -> None:
        """Synchronizes parameters of training model to inference model."""
        ...

    @property
    def need_sync(self) -> bool:
        """Return whether It is necessary to synchronize training model and
        inference model."""
        return self.has_inference_model and (not self.inference_only)
```